### DISCUSSION TITLE
[HxButton] using Tooltip never fills available width

### DISCUSSION THREAD

**Discussion by bholbrook** [havit/Havit.Blazor#862] [Q&A]
Adding the ```Tooltip``` property to a ```HxButton``` prevents the button from filling the available width.

Working example without Tooltip:
``` html
<div class="d-flex">
    <HxButton Color="ThemeColor.Primary" Icon="BootstrapIcon.EmojiSmile" CssClass="flex-grow-1" />
</div>
</style>
```

Non-working example with Tooltip:
``` html
<div class="d-flex">
    <HxButton Color="ThemeColor.Primary" Icon="BootstrapIcon.EmojiSmile" CssClass="flex-grow-1" Tooltip="Hello" />
</div>
</style>
```

![image](https://github.com/user-attachments/assets/3455fd9a-03b8-476a-9db4-87234b96af93)

Is a fix for this possible? Or perhaps even a workaround?

**Comment by crdo** [comment]
If you use the `tooltip` built-in to the button, the `button` gets wrapped in another `span` that is accessible via `TooltipWrapperCssClass`. 

The reason for this is because we wanted the tooltips to work even for disabled buttons (so you can explain users why is it disabled).

Therefore you need to do the `flexbox` magic on this span too (btw. if you use `.d-grid` it is more straightforward as you only need to set the class on the container).

It is documented here https://havit.blazor.eu/components/HxButton#tooltip-wrapper-and-css-classes
